cmd/serve.go

```diff
@@ -21,18 +21,22 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"math"
 	"net"
 	"net/http"
+	"net/url"
 	"sync"
 	"sync/atomic"
 	"time"
 
 	"github.com/dustin/go-humanize"
 	"github.com/minio/bottlenet/pkg/perf"
+
+	utilio "gopkg.in/src-d/go-git.v4/utils/ioutil"
```

Comment:
This package is archived @wlan0 do you really want to use it? why depend on something that is unmaintained. 

It is worth to just write a simple implementation like this

```
// Returns error if the cancelCh has been closed (indicating that S3 client has disconnected)                                                          
 type detectDisconnect struct {                                                                                                                         
         io.ReadCloser                                                                                                                                  
         cancelCh <-chan struct{}                                                                                                                       
 }                                                                                                                                                      
                                                                                                                                                        
 func (d *detectDisconnect) Read(p []byte) (int, error) {                                                                                               
         select {                                                                                                                                       
         case <-d.cancelCh:                                                                                                                             
                 return 0, io.ErrUnexpectedEOF                                                                                                          
         default:                                                                                                                                       
                 return d.ReadCloser.Read(p)                                                                                                            
         }                                                                                                                                              
 }                
```

Do you see a problem with this code? 

you can even use direct context

```
// Returns error if the cancelCh has been closed (indicating that S3 client has disconnected)                                                          
 type detectDisconnect struct {                                                                                                                         
         io.ReadCloser                                                                                                                                  
         ctx context.Context                                                                                                             
 }                                                                                                                                                      
                                                                                                                                                        
 func (d *detectDisconnect) Read(p []byte) (int, error) {                                                                                               
         select {                                                                                                                                       
         case <-d.ctx.Done():                                                                                                                             
                 return 0, d.ctx.Err()                                                                                                         
         default:                                                                                                                                       
                 return d.ReadCloser.Read(p)                                                                                                            
         }                                                                                                                                              
 }      
```

Reply:
Good point, yeah its not complex. I'll update it